### PR TITLE
File creation fixed (Issue  #103)

### PR DESCRIPTION
**Pull request by Kuvrot**
### Fix to the problem where the files were being created only at the root folder instead of the desired one (Issue #103 )

https://github.com/michaelsakharov/Prowl/assets/23508114/8ba3b94d-ce69-4880-802a-e70ce10895cd

I made the CurDirectory variable from the AssetBrowserWindow.cs static in order to indicate the path where the file will be created in MainMenuItems.cs

### CODE REVIEW

File: Prowl.Editor/EditorWindows/MainMenuItems.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ public static class MainMenuItems
         [MenuItem("Create/Folder")]
         public static void CreateFolder()
         {
-            Directory ??= new DirectoryInfo(Project.ProjectAssetDirectory);
```
Comment (michaelsakharov) [inline Prowl.Editor/EditorWindows/MainMenuItems.cs:19]: Hmm, these null checks here were placed as a fallback when no AssetBrowserWindow has been created.
AssetBrowserWindow.CurDirectory can be null
It also should support when right-clicking a folder and pressing Create Material, it should create it inside the folder not the directory I'm inside.

Comment (michaelsakharov) [reply Prowl.Editor/EditorWindows/MainMenuItems.cs:19]: The problem is further up, Whatever is calling MenuItem.DrawMenuRoot("Create");
Isnt setting the MainMenuItems.Directory properly, which at a first glance looks like thats the case in a few issues.

**Comment by michaelsakharov** [comment]
Pushed a possible fix to this just now, Let me know if it works.

File: Prowl.Editor/EditorWindows/MainMenuItems.cs (hunk 1 above)
Comment (Kuvrot) [reply Prowl.Editor/EditorWindows/MainMenuItems.cs:19]: okok, I get it now.

**Comment by Kuvrot** [comment]
I didn't work for me.

**Comment by michaelsakharov** [comment]
Dang, Can you show a clip of how exactly your creating the files?

**Comment by michaelsakharov** [comment]
Sorry i missed a compile error

**Comment by Kuvrot** [comment]
> It also should support when right-clicking a folder and pressing Create Material, it should create it inside the folder not the directory I'm inside.

To be more specific, this is working, right clicking a folder and creating a file, will create the file inside the folder clicked.

What is not working is this: 

https://github.com/michaelsakharov/Prowl/assets/23508114/bc0b4880-354c-418c-ad73-236094a462d8

**Comment by michaelsakharov** [comment]
> > It also should support when right-clicking a folder and pressing Create Material, it should create it inside the folder not the directory I'm inside.
> 
> To be more specific, this is working, right clicking a folder and creating a file, will create the file inside the folder clicked.
> 
> What is not working is this:
> 
>  Prowl2024-03-0217-41-29-ezgif.com-crop-video.mp4

This seems to be working for me on my last commit

**Comment by Kuvrot** [comment]
Oh really?, let me check it again then.

**Comment by Kuvrot** [comment]
Ahhhh yes, sorry you are right, I made a new clone of the repo and it's working.

**Comment by michaelsakharov** [comment]
Awesome :D

**Pull request closed without merging**